src/core/coap/coap.cpp

```diff
@@ -129,25 +146,76 @@ otError CoapBase::Send(ot::Message &aMessage, const Ip6::MessageInfo &aMessageIn
     return mSender(*this, aMessage, aMessageInfo);
 }
 
+#if OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
+otError CoapBase::SendMessage(Message &                   aMessage,
+                              const Ip6::MessageInfo &    aMessageInfo,
+                              const TxParameters &        aTxParameters,
+                              ResponseHandler             aHandler,
+                              void *                      aContext,
+                              otCoapBlockwiseTransmitHook aTransmitHook,
+                              otCoapBlockwiseReceiveHook  aReceiveHook)
+#else
 otError CoapBase::SendMessage(Message &               aMessage,
                               const Ip6::MessageInfo &aMessageInfo,
                               const TxParameters &    aTxParameters,
                               ResponseHandler         aHandler,
                               void *                  aContext)
+#endif
 {
     otError  error;
     Message *storedCopy = nullptr;
     uint16_t copyLength = 0;
+#if OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
+    uint8_t  buf[kMaxBlockLength] = {0};
```

Comment:
Do we need the buffer to initialized to zero? Seems like we would be reading it from message below.
If not I would suggest removing `{0}`?